core/src/main/java/org/apache/accumulo/core/clientImpl/ClientTabletCacheImpl.java

```diff
@@ -692,64 +640,136 @@ private void requestTabletHosting(ClientContext context,
     }
   }
 
-  private void lookupTablet(ClientContext context, Text row, boolean retry,
-      LockCheckerSession lcSession) throws AccumuloException, AccumuloSecurityException,
-      TableNotFoundException, InvalidTabletHostingRequestException {
+  private static class ReferenceCountedLock {
+    final Lock lock = new ReentrantLock();
+    int refCount;
```

Comment:
Should this be volatile?

Reply:
was assuming it was modified in a synchronized block or a lock in the maps compute function and would not need to be volatile, but I need to verify that assumption